contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -645,13 +790,21 @@ abstract contract PoSValidatorManager is
             return 0;
         }
 
-        return $._rewardCalculator.calculateReward({
+        uint256 reward = $._rewardCalculator.calculateReward({
             stakeAmount: weightToValue(delegator.weight),
             validatorStartTime: validator.startedAt,
             stakingStartTime: delegator.startedAt,
             stakingEndTime: delegationEndTime,
             uptimeSeconds: $._posValidatorInfo[delegator.validationID].uptimeSeconds
         });
+
+        $._redeemableDelegatorRewards[delegationID] = reward;
+
+        if (rewardRecipient != address(0)) {
+            $._delegatorRewardRecipients[delegationID] = rewardRecipient;
+        }
```

Comment:
Would it maybe make sense to set this in `initializeEndDelegation`, so that we don't have to pass the parameter down another level?

Reply:
I actually just changed this because of [this comment](https://github.com/ava-labs/teleporter/pull/647#discussion_r1852348458). 

Is there a problem passing this down? More expensive? I can revert my change, but this definitely is cleaner (ignoring gas cost)

Reply:
Yeah no, this is good :)